multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/StageAppStep.java

```diff
@@ -46,7 +47,7 @@ protected List<AsyncExecution> getAsyncStepExecutions(ProcessContext context) {
 
     @Override
     public Duration getTimeout(ProcessContext context) {
-        return context.getVariable(Variables.START_TIMEOUT);
+        return getGivenTimeout(context, SupportedParameters.STAGE_TIMEOUT, Variables.STAGE_APP_TIMEOUT, Variables.STAGE_APP_TIMEOUT_GLOBAL,
+                               Messages.STAGE_APP_TIMEOUT);
```

Comment:
the message doesn't need to be specific to the type of timeout - can make it generic and insert the specific timeout variable name upon logging it in the 'getGivenTimeout' method